src/codeu/chat/server/Server.java

```diff
@@ -58,20 +61,57 @@ public Server(Uuid id, byte[] secret, Relay relay) {
 
     this.controller = new Controller(id, model);
     this.relay = relay;
-  }
 
-  public void syncWithRelay(int maxReadSize) throws Exception {
-    for (final Relay.Bundle bundle :  relay.read(id, secret, lastSeen, maxReadSize)) {
-      onBundle(bundle);
-      lastSeen = bundle.id();
-    }
+    timeline.scheduleNow(new Runnable() {
+      @Override
+      public void run() {
+        try {
+
+          LOG.info("Reading update from relay...");
+
+          for (final Relay.Bundle bundle : relay.read(id, secret, lastSeen, 32)) {
+            onBundle(bundle);
+            lastSeen = bundle.id();
+          }
+
+        } catch (Exception ex) {
+
+          LOG.error(ex, "Failed to read update from relay.");
```

Comment:
Short answer: The stack trace is printed last.

Long answer : All logging functions can specify a string followed by format values: LOG.info("Something as called %d times", 5); But for error we wanted the stack trace, but to keep the same pattern as the other log functions, the exception was put as the first parameter. How it is printed is handled by the logging system behind the scenes (part of Java) and not something we control directly.